components/scheduler2/scheduler.go

```diff
@@ -0,0 +1,226 @@
+package scheduler2
+
+import (
+	"fmt"
+	"time"
+
+	"go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/schedule/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	schedspb "go.temporal.io/server/api/schedule/v1"
+	"go.temporal.io/server/service/history/hsm"
+	"go.temporal.io/server/service/worker/scheduler"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	// Scheduler is a top-level state machine compromised of 3 sub state machines:
+	// - Generator: buffers actions according to the schedule specification
+	// - Executor: executes buffered actions
+	// - Backfiller: buffers actions according to requested backfills
+	//
+	// A running Scheduler will always have exactly one of each of the above sub state
+	// machines mounted as nodes within the HSM tree. The top-level	machine itself
+	// remains in a singular running state for its lifetime (all work is done within the
+	// sub state machines). The Scheduler state machine is only responsible for creating
+	// the singleton sub state machines.
+	Scheduler struct {
+		*schedspb.SchedulerInternal
```

Comment:
BTW, you could save a pointer and replace this with `schedspb.SchedulerInternal`, I regret not doing the same in the other state machines.